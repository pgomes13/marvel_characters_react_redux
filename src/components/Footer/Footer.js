import React from 'react';
import PropTypes from 'prop-types';
import Typography from "@material-ui/core/Typography/Typography";
import { withStyles } from '@material-ui/core/styles';

import { FooterStyles } from './Footer.styles';

/**
 * The Footer component consisting of the footer info
 * @param {Object} classes - the material-ui classes prop
 * @returns {Node} - the Footer component
 */
const _Footer = ({ classes, title, content, html }) => (
    <footer className={classes.footer}>
        <Typography variant="h6" align="center" gutterBottom>
            {title}
        </Typography>
        {content &&
        <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
            {content}
        </Typography>
        }
        {html &&
        <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
            <div dangerouslySetInnerHTML={{ __html: html }} />
        </Typography>
        }

    </footer>
);

_Footer.propTypes = {
    classes: PropTypes.object.isRequired,
    title: PropTypes.string,
    content: PropTypes.string
};

const Footer = withStyles(FooterStyles)(_Footer);

export { Footer };
