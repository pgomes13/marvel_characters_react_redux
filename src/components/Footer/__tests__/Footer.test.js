import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { _Footer } from '../Footer';

describe('Footer', () => {
    const props = {
        classes: {},
        imageUrl: 'http://test.test.com',
        title: 'test',
        description: 'test'
    };

    it('component renders correctly with props', () => {
        const wrapper = shallow(<_Footer {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});