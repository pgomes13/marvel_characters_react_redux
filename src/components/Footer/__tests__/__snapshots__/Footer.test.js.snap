// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders correctly with props 1`] = `
<footer>
  <WithStyles(Typography)
    align="center"
    gutterBottom={true}
    variant="h6"
  >
    test
  </WithStyles(Typography)>
</footer>
`;
